xds/internal/xdsclient/channel.go

```diff
@@ -287,7 +287,7 @@ func decodeResponse(opts *xdsresource.DecodeOptions, rType xdsresource.Type, res
 		perResourceErrors[name] = err
 		// Add place holder in the map so we know this resource name was in
 		// the response.
-		ret[name] = ads.DataAndErrTuple{Err: err}
+		ret[name] = ads.DataAndErrTuple{Err: xdsresource.NewErrorf(xdsresource.ErrorTypeNACKed, "%s", err.Error())}
```

Comment:
Totally optional. But all instances of the error type `ErrorTypeNACKed` have the same format string `%s` which is always set to `err.Error()` or a new error string. What if we add a function to the `xdsresource` package that is:

```
func NewError(t ErrorType, err string) error { ... }
```

So, callsites can either be `xdsresource.NewError(xdsresource.ErrorTypeNACKed, err.Error())` or `xdsresource.NewError(xdsresource.ErrorTypeNACKed, "some new error string")`

Reply:
Created a new function named `NewError` and changed callers of `NewErrorf` that were not using format strings.